### PR TITLE
DOC-858 Add follow_redirects field

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves [DOC-858](https://redpandadata.atlassian.net/browse/DOC-858)
Review deadline: 12th December

## Page previews

[`http` processor](https://deploy-preview-129--redpanda-connect.netlify.app/redpanda-connect/components/processors/http/)

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

[DOC-858]: https://redpandadata.atlassian.net/browse/DOC-858?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/components/pages/processors/http.adoc [hunk 1]
```diff
@@ -105,11 +106,15 @@ In order to map or encode the payload to a specific request body, and map the re
 
 == Response codes
 
-Redpanda Connect considers any response code between 200 and 299 inclusive to indicate a successful response, you can add more success status codes with the field `successful_on`.
```
Comment (asimms41) [inline modules/components/pages/processors/http.adoc:108]: Please check this section for accuracy. I've added a new line about redirects and refactored the rest of the paragraph so it is (hopefully) easier to read.

**Review by ooesili — APPROVED** [review]

File: modules/components/pages/processors/http.adoc [hunk 2]
```diff
@@ -105,11 +106,15 @@ In order to map or encode the payload to a specific request body, and map the re
...
+HTTP response codes in the 200-299 range indicate a successful response. You can use the <<successful_on,`successful_on`>> field to add more success status codes.
```
Comment (Deflaimun) [inline modules/components/pages/processors/http.adoc:109]: Another way to convey this information is by saying 2xx. It's a more common way to http codes.

 https://sitechecker.pro/what-are-2xx-status-codes/

**Review by Deflaimun — APPROVED** [review]

File: modules/components/pages/processors/http.adoc [hunk 3]
```diff
@@ -741,17 +742,24 @@ The maximum period to wait between failed requests.
...
 The maximum number of retry attempts to make.
 
-
 *Type*: `int`
 
 *Default*: `3`
 
+=== `follow_redirects`
+
+Whether to follow redirects, including all responses with HTTP status codes in the 300-399 range. If set to `false`, the response message includes only the body, status, and headers from the redirect response, and this processor does not make a request to the URL specified in the Location header.
```
Comment (Deflaimun) [inline modules/components/pages/processors/http.adoc:757]:
```suggestion
Whether to follow redirects, including all responses with HTTP status codes in the 300-399 range. If set to `false`, the response message includes only the body, status, and headers from the redirect response, and this processor does not make a request to the URL specified in the location header.
```

Comment (Deflaimun) [inline modules/components/pages/processors/http.adoc:757]:
```suggestion
Whether to follow redirects, including all responses with HTTP status codes in the 300-399 range. If set to `false`, the response message includes only the body, status, and headers from the redirect response, and this processor does not make a request to the URL specified in the `Location` header.
```

File: modules/components/pages/processors/http.adoc (hunk 2 above)
Comment (asimms41) [reply modules/components/pages/processors/http.adoc:109]: Will leave for now. My reasoning is that it's just an easy way to scan the values without having to look them.

**Pull request merged**